data_structures/k_tree/k_tree.py

```diff
@@ -0,0 +1,81 @@
+class Node:
+    def __init__(self, val, children=[]):
+        self.val = val
+        self.children = []
+        for child in children:
+            self.children.append(Node(child))
+
+    def __repr__(self):
+        return 'node is {}'.format(str(self.val))
+
+    def __str__(self):
+        return str(self.val)
+
+
+class KTree:
+    def __init__(self, val):
+        self.root = Node(val)
+
+    def __repr__(self):
+        return 'root is {}'.format(str(self.root.val))
+
+    def __str__(self):
+        return str(self.root.val)
+
+    def insert(self, val, parent):
+        self.breadth_first_traversal(lambda x: x.children.append(Node(val)) if x.val == parent else False)
+
+    def in_order(self, operation):
```

Comment:
This is not a method you _can_ implement for the k-ary tree because each node can have any number of children.